src/containers/SubjectPage/components/SubjectPageContent.jsx

```diff
@@ -61,68 +49,58 @@ const SubjectPageContent = ({
           onClickTopics(e);
         }}
       />
-      {topicId && (
-        <div ref={mainRef}>
-          <MainTopic
-            topicId={topicId}
-            subjectId={subject.id}
-            filterIds={filterIds}
-            setSelectedTopic={setSelectedTopic}
-            showResources={!subTopicId}
-            subTopicId={subTopicId}
-            locale={locale}
-            ndlaFilm={ndlaFilm}
-            onClickTopics={onClickTopics}
-          />
-        </div>
-      )}
-      {subTopicId && (
-        <div ref={subRef}>
-          <SubTopic
-            topicId={subTopicId}
-            subjectId={subject.id}
-            filterIds={filterIds}
-            setSelectedSubTopic={setSelectedSubTopic}
-            locale={locale}
-            ndlaFilm={ndlaFilm}
-            subSubTopicId={subSubTopicId}
-            onClickTopics={onClickTopics}
-          />
-        </div>
-      )}
-      {subSubTopicId && (
-        <div ref={subSubRef}>
-          <SubTopic
-            topicId={subSubTopicId}
-            subjectId={subject.id}
-            filterIds={filterIds}
-            setSelectedSubTopic={setSelectedSubSubTopic}
-            locale={locale}
-            ndlaFilm={ndlaFilm}
-            onClickTopics={onClickTopics}
-          />
-        </div>
-      )}
+      {topics.map((t, index) => {
+        if (index === 0) {
+          return (
+            <div ref={refs[index]}>
+              <MainTopic
+                topicId={topics[0]}
+                subjectId={subject.id}
+                filterIds={filterIds}
+                setBreadCrumb={setBreadCrumb}
+                showResources={!topics[index + 1]}
+                subTopicId={topics[index + 1]}
+                locale={locale}
+                ndlaFilm={ndlaFilm}
+                onClickTopics={onClickTopics}
+                index={index}
+              />
+            </div>
+          );
+        } else {
+          return (
+            <div ref={refs[index]}>
+              <SubTopic
+                topicId={t}
+                subjectId={subject.id}
+                filterIds={filterIds}
+                setBreadCrumb={setBreadCrumb}
+                locale={locale}
+                ndlaFilm={ndlaFilm}
+                subSubTopicId={
+                  topics[index + 1] ? topics[index + 1] : undefined
+                }
+                onClickTopics={onClickTopics}
+                index={index}
+                topics={topics}
```

Comment:
Ser ikkje at denne verdien benyttes i SubTopic komponenten. Til gjengjeld benyttes den som parameter til subSubTopicId nokre linjer over her. Kunne disse to blitt slått sammen? At subSubTopicId regnes ut inne i subTopic basert på topics og index og ikkje sendes inn som en parameter?

Alternativt at topics som sendes inn til en SubTopic kun inneheld dei topics som faktisk er subtopics til det som rendres. Altså en topics={topics.slice(1)}